### PR TITLE
certificates migrations

### PR DESCRIPTION
**Pull request by Dinek007**

### CODE REVIEW

File: src/renderer/store/selectors/contacts.ts [hunk 1]
```diff
@@ -198,30 +198,35 @@ const allChannels = createSelector(
...
       acc[parsedCerficated] = {
         username: nickname,
         onionAddress: onionAddress,
-        peerId: peerId
+        peerId: peerId,
+        dmPubKey: dmPubKey
       }
       return acc
     }, {})
+    if (accumulator) {
```
Comment (kowalski) [inline src/renderer/store/selectors/contacts.ts:225]: this is always at least `{}` so truthy

this whole `if-else` is redundant

File: src/renderer/store/certificates/certificates.saga.ts [hunk 2]
```diff
@@ -55,10 +56,13 @@ export function* createOwnCertificate(
...
   }
 
+  const dmPublicKey = yield* select(directMessagesSelectors.publicKey)
+
   const userData = {
     zbayNickname: action.payload,
     commonName: hiddenServices.libp2pHiddenService.onionAddress,
-    peerId: peerIdAddress
+    peerId: peerIdAddress,
+    dmPublicKey: hexStringToArrayBuffer(dmPublicKey)
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:65]: `hexStringToArrayBuffer` should be used inside `@zbay/identity`, here we should pass hex string

**Review by kowalski — CHANGES_REQUESTED** [review]
see inline comments

File: src/renderer/store/certificates/certificates.saga.ts [hunk 3]
```diff
@@ -50,27 +55,36 @@ export function* createOwnCertificate(
...
+    peerId: peerIdAddress,
+    dmPublicKey: dmPublicKey,
+    signAlg: configCrypto.signAlg,
+    hashAlg: configCrypto.hashAlg
   }
 
   console.log('userData', userData)
 
   const user = yield* call(createUserCsr, userData)
+  console.log(user)
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:79]: console.log

**Review by kowalski — APPROVED** [review]

File: src/renderer/sagas/socket/socket.saga.ts [hunk 4]
```diff
@@ -319,7 +319,22 @@ export function* responseGetCertificates(socket: Socket): Generator {
 export function* addCertificate(): Generator {
   const hasCertyficate = yield* select(certificatesSelectors.ownCertificate)
   const nickname = yield* select(identitySelectors.nickName)
-  if (!hasCertyficate && nickname) {
+  let parsedCert
+  let updateCertificate = false
+
+  if (hasCertyficate) {
```
Comment (holmesworcester) [inline src/renderer/sagas/socket/socket.saga.ts:325]: Confusing spelling: "Certyficate"

**Pull request merged**